web/src/inbox_ui.js

```diff
@@ -0,0 +1,875 @@
+import * as compose_closed_ui from "./compose_closed_ui";
+import * as hash_util from "./hash_util";
+import * as hashchange from "./hashchange";
+import {is_visible, set_visible} from "./inbox_util";
+import * as keydown_util from "./keydown_util";
+import * as left_sidebar_navigation_area from "./left_sidebar_navigation_area";
+import {localstorage} from "./localstorage";
+import * as message_store from "./message_store";
+import * as message_view_header from "./message_view_header";
+import * as narrow from "./narrow";
+import * as narrow_state from "./narrow_state";
+import * as navigate from "./navigate";
+import * as people from "./people";
+import * as pm_list from "./pm_list";
+import * as search from "./search";
+import * as stream_color from "./stream_color";
+import * as stream_data from "./stream_data";
+import * as stream_list from "./stream_list";
+import * as sub_store from "./sub_store";
+import * as unread from "./unread";
+import * as unread_ops from "./unread_ops";
+import * as unread_ui from "./unread_ui";
+import * as util from "./util";
+
+let pms_dict = {};
+let topics_dict = {};
+let streams_dict = {};
+let include_muted = false;
+let update_triggered_by_user = false;
+
+const COLUMNS = {
+    COLLAPSE_BUTTON: 0,
+    RECIPIENT: 1,
+    UNREAD_COUNT: 2,
+};
+let col_focus = COLUMNS.RECIPIENT;
+let row_focus = 0;
+
+const ls_key = "inbox_mute_filter";
+const ls = localstorage();
+let search_keyword = "";
+const INBOX_SEARCH_ID = "inbox_search";
+const MUTED_FILTER_ID = "include_muted";
+export let current_focus_id = INBOX_SEARCH_ID;
+
+const STREAM_HEADER_PREFIX = "stream_header_";
+const CONVERSATION_ID_PREFIX = "inbox_";
+
+function get_row_from_conversation_key(key) {
+    return $(`#${CONVERSATION_ID_PREFIX}` + CSS.escape(`${key}`));
+}
+
+function save_filter() {
+    ls.set(ls_key, include_muted);
+}
+
+export function show() {
+    if (narrow.has_shown_message_list_view) {
+        narrow.save_pre_narrow_offset_for_reload();
+    }
+
+    if (is_visible()) {
+        return;
+    }
+
+    left_sidebar_navigation_area.narrow_to_inbox();
+    stream_list.handle_narrow_deactivated();
+
+    $("#message_feed_container").hide();
+    $("#inbox_view").show();
+    set_visible(true);
+
+    unread_ui.hide_unread_banner();
+    narrow_state.reset_current_filter();
+    message_view_header.render_title_area();
+    narrow.handle_middle_pane_transition();
+
+    narrow_state.reset_current_filter();
+    narrow.update_narrow_title(narrow_state.filter());
+    message_view_header.render_title_area();
+    pm_list.handle_narrow_deactivated();
+    search.clear_search_form();
+    complete_rerender();
+    compose_closed_ui.set_standard_text_for_reply_button();
+}
+
+export function hide() {
+    const $focused_element = $(document.activeElement);
+
+    if ($("#inbox_view").has($focused_element)) {
+        $focused_element.trigger("blur");
+    }
+
+    $("#message_feed_container").show();
```

Comment:
Similarly, is it correct to show the message feed container on our way to recent topics? I guess it might work in that`recent_topics.show` function will hide it again?

Reply:
Thinking about this, I think it's possible we can merge this version, but we might want to be thinking about some refactor that would create a common "Switch center pane to X" function, perhaps with some callbacks, that doesn't end up with each of this, `narrow.js`, and `recent_topics_ui.js` have a bunch of duplicated code. I don't have a clear idea of how this should work, and it might be a fine follow-up refactor, but maybe you have an idea that would be nice here.

Reply:
Okay, added to TODO.

Reply:
Cool, feel free to put it into an issue in the inbox view label if that feels helpful for tracking.

Reply:
Opened #26688